Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -633,17 +633,25 @@ extension HTTPClient {
 
         let promise: EventLoopPromise<Response>
         var completion: EventLoopFuture<Void>
-        var connection: Connection?
-        var cancelled: Bool
-        let lock: Lock
+        private let lock = Lock()
+        // protected by lock
+        private var _connection: Connection?
+        // protected by lock
+        private var _cancelled: Bool = false
         let logger: Logger // We are okay to store the logger here because a Task is for only one request.
 
+        var isCancelled: Bool {
+            self.lock.withLock { self._cancelled }
+        }
+
+        var connection: Connection? {
```

Comment:
It seems like this is only used in tests: can we mark it clearly for that purpose?

Reply:
Will do!

Reply:
Sorry, I wasn't clear: I meant changing the name to `var _testOnly_connection`

Reply:
This is used on L715, maybe make it private and add an internal `_testOnly_connection` per Cory's suggestion?